statics/js/components/preference.js

```diff
@@ -149,7 +150,52 @@ var PreferenceComponent = {
         if (f.name !== "" && f.filter !== "") newList.push(f);
       });
       return newList;
-    }
+    },
+
+    saveToFile: function() {
+      var obj = {};
+      obj["filters"] = JSON.stringify(this.filterEmpty(this.filters));
```

Comment:
Is there a way to iterate on the page so we don't need to add specific 'key' for each field, can you tag them in the html and iterate on all tags .... this will allow to add more and more parameters without need to change the save and load functions everytime

Reply:
+1
we could simply add a "preferences" field to "this" like 

JSON.stringify(this.preferences)

and preferences would be like this 

```
this.preferences = {
  bandwidthThreshold:
  ...
```

Reply:
done